### PR TITLE
soc: nxp_imx: use zephyr,flash node for flash region size and offset

### PR DESCRIPTION
**Pull request by danieldegrasse**
This patch enables NXP iMX RT boards to use the zephyr flash node for the flash region size and offset, instead of using `CONFIG_CODE_LOCATION` choice to set current code location. It also corrects flash sizes for some boards, to enable a Zephyr image to use the entire flash region. The following changes are included in this patch:

- update board level flash part definitions to encode the offset of the memory mapped flexspi region, as well as the size of the flash part.
- remove the `CONFIG_CODE_LOCATION` choice from SOC level KConfig files for iMX RT SOCs, as well as any overrides to `CONFIG_FLASH_SIZE` and `CONFIG_FLASH_BASE_ADDRESS`, so the default definitions in `zephyr/arch/KConfig` will be used.
- update the flexspi dts bindings to allow encoding a register address and size, since most child nodes of spi devices do not encode this in their `reg` property. Additionally, suppress the DTS compiler warning that encoding the flash address and size generates
- update board level DTS files to select the flash part as the `zephyr,flash` chosen node, as well as the `zephyr,flash-controller` node

*Note: this PR depends on the following two PRs:*
#44760 for support for larger flash MPU region sizes
#44759 for the `dt_chosen_has_compat` helper function, which is used to determine if the flash boot header should be added to the image.

### CODE REVIEW

File: boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts [hunk 1]
```diff
@@ -52,11 +52,12 @@ arduino_serial: &lpuart1 {};
 
 &flexspi {
 	reg = <0x402a8000 0x4000>, <0x60000000 DT_SIZE_M(16)>;
-	at25sf128a: at25sf128a@0 {
+	at25sf128a: at25sf128a@60000000 {
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts:56]: I tried this before and got a NAK from @galak; he felt strongly that this address should be the SPI chip select. See #30065. I really would like to get rid of the CODE_LOCATION stuff though.

Comment (galak) [reply boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts:56]: Yeah, I probably have the same NAK as before.

Comment (danieldegrasse) [reply boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts:56]: See my comment below- I agree with @MaureenHelm that there would be real value to removing the `CODE_LOCATION` Kconfig values, and I'm happy to change the approach we are taking here to preserve the SPI chip select values. The main priority is enabling the use of `zephyr,flash` to select the code location, to simplify changing the code location on our platforms for customers.

**Comment by dleach02** [comment]
@galak, can you provide guidance on what would clear the NAK?

**Pull request closed without merging**